remote/remote.go

```diff
@@ -186,18 +186,24 @@ func newV1Image(keychain authn.Keychain, repoName string, platform imgutil.Platf
 		OSVersion:    platform.OSVersion,
 	}
 
-	image, err := remote.Image(ref, remote.WithAuth(auth), remote.WithTransport(http.DefaultTransport), remote.WithPlatform(v1Platform))
-	if err != nil {
-		if transportErr, ok := err.(*transport.Error); ok && len(transportErr.Errors) > 0 {
-			switch transportErr.StatusCode {
-			case http.StatusNotFound, http.StatusUnauthorized:
+	var image v1.Image
+	for i := 0; i < 3; i++ {
+		image, err = remote.Image(ref, remote.WithAuth(auth), remote.WithTransport(http.DefaultTransport), remote.WithPlatform(v1Platform))
+		if err != nil {
+			if err == io.EOF {
+				continue // retry if EOF
```

Comment:
Given the nature of the linked comment, I'm wondering if we should introduce a little delay between attempts.

Reply:
Added in https://github.com/buildpacks/imgutil/pull/96/commits/36099cee419a40551bac14161c75e57b3b0ce293